signingscript/task.py

```diff
@@ -142,203 +114,55 @@ def get_suitable_signing_servers(signing_servers, cert_type, signing_formats):
-async def _convert_dmg_to_tar_gz(context, from_):
-    """Explode a dmg and tar up its contents. Return the relative tarball path."""
-    work_dir = context.config['work_dir']
-    abs_from = os.path.join(work_dir, from_)
-    # replace .dmg suffix with .tar.gz (case insensitive)
-    to = re.sub('\.dmg$', '.tar.gz', from_, flags=re.I)
-    abs_to = os.path.join(work_dir, to)
-    dmg_executable_location = context.config['dmg']
-    hfsplus_executable_location = context.config['hfsplus']
-
-    with tempfile.TemporaryDirectory() as temp_dir:
-        app_dir = os.path.join(temp_dir, "app")
-        utils.mkdir(app_dir)
-        temp_dir = os.path.join(os.getcwd(), "tmp")
-        undmg_cmd = [dmg_executable_location, "extract", abs_from, "tmp.hfs"]
-        await utils._execute_subprocess(undmg_cmd, cwd=temp_dir)
-        hfsplus_cmd = [hfsplus_executable_location, "tmp.hfs", "extractall", "/", app_dir]
-        await utils._execute_subprocess(hfsplus_cmd, cwd=temp_dir)
-        tar_cmd = ['tar', 'czvf', abs_to, '.']
-        await utils._execute_subprocess(tar_cmd, cwd=app_dir)
-
-    return to
-
-
-# _extract_zipfile {{{1
-async def _extract_zipfile(context, from_, tmp_dir=None):
-    work_dir = context.config['work_dir']
-    tmp_dir = tmp_dir or os.path.join(work_dir, "unzipped")
-    try:
-        files = []
-        rm(tmp_dir)
-        utils.mkdir(tmp_dir)
-        with zipfile.ZipFile(from_, mode='r') as z:
-            for name in z.namelist():
-                files.append(os.path.join(tmp_dir, name))
-            z.extractall(path=tmp_dir)
-        return files
-    except Exception as e:
-        raise SigningScriptError(e)
-
-
-# _create_zipfile {{{1
-async def _create_zipfile(context, to, files, tmp_dir=None):
-    work_dir = context.config['work_dir']
-    tmp_dir = tmp_dir or os.path.join(work_dir, "unzipped")
-    try:
-        with zipfile.ZipFile(to, mode='w', compression=zipfile.ZIP_DEFLATED) as z:
-            for f in files:
-                relpath = os.path.relpath(f, tmp_dir)
-                z.write(f, arcname=relpath)
-        return to
-    except Exception as e:
-        raise SigningScriptError(e)
-
-
-# detached_sigfiles {{{1
-def detached_sigfiles(filepath, signing_formats):
-    """Get a list of detached sigfile paths, if any, given a file path and signing formats.
-
-    This will generally be an empty list unless we're gpg signing, in which case
-    we'll have detached gpg signatures.
+    output = path
+    # Loop through the formats and sign one by one.
+    for fmt in signing_formats:
+        func = FORMAT_TO_SIGNING_FUNCTION.get(
+            fmt, FORMAT_TO_SIGNING_FUNCTION['default']
+        )
+        log.info("Signing {} with {}...".format(output, fmt))
+        output = await func(context, output, fmt)
+    # We want to return a list
+    if not isinstance(output, (tuple, list)):
+        output = [output]
+    # Log the shas of each output file
+    log_shas(context, output)
```

Comment:
I think this is leftover from funsize signing worker: https://github.com/mozilla-releng/signingscript/commit/d827a74a4d77379ee104c853484e31ba7667ae83